core/src/main/java/org/apache/accumulo/core/metadata/schema/SelectedFiles.java

```diff
@@ -63,23 +72,63 @@ public SelectedFiles(Set<StoredTabletFile> files, boolean initiallySelectedAll,
     jData.selAll = initiallySelectedAll;
     // ELASITICITY_TODO need the produced json to always be the same when the input data is the same
     // as its used for comparison. Need unit test to ensure this behavior.
-    metadataValue = GSON.get().toJson(jData);
+    metadataValue = GSON.toJson(jData);
   }
 
-  private SelectedFiles(Set<StoredTabletFile> files, boolean initiallySelectedAll, long fateTxId,
-      String metaVal) {
-    Preconditions.checkArgument(files != null && !files.isEmpty());
-    this.files = files;
-    this.initiallySelectedAll = initiallySelectedAll;
-    this.fateTxId = fateTxId;
-    this.metadataValue = metaVal;
+  private static class SelectedFilesTypeAdapter extends TypeAdapter<SelectedFiles> {
+
+    @Override
+    public void write(JsonWriter out, SelectedFiles selectedFiles) throws IOException {
+      out.beginObject();
+      out.name("txid").value(FateTxId.formatTid(selectedFiles.getFateTxId()));
+      out.name("selAll").value(selectedFiles.initiallySelectedAll());
+      List<String> sortedFiles = selectedFiles.getFiles().stream()
+          .map(StoredTabletFile::getMetaUpdateDelete).sorted().collect(Collectors.toList());
+      out.name("files").beginArray();
+      for (String file : sortedFiles) {
+        out.value(file);
+      }
+      out.endArray();
```

Comment:
Could avoid creating the list w/ something like the following.

```suggestion
      out.name("files").beginArray();
      // sort the data to make serialized json comparable
     selectedFiles.getFiles().stream()
          .map(StoredTabletFile::getMetaUpdateDelete).sorted().forEach(out::value);
      out.endArray();
```